JiraPS/Public/Remove-JiraVersion.ps1

```diff
@@ -10,60 +11,75 @@
     .EXAMPLE
        Remove-JiraVersion -ID '66596'
        This example removes the Version given.
-    .INPUTS
-       This function does not accept pipeline input pending class creation.
+     .INPUTS
+        [PSJira.Versions]
     .OUTPUTS
        This Function outputs no results
     .NOTES
        This function requires either the -Credential parameter to be passed or a persistent JIRA session. See New-JiraSession for more details.  If neither are supplied, this function will run with anonymous access to JIRA.
     #>
-    [CmdletBinding(DefaultParameterSetName = 'Project')]
+    [CmdletBinding(DefaultParameterSetName = 'VersionID')]
     param(
         # Jira Version Name
         [Parameter(Mandatory = $true,
-            ParameterSetName = 'Project')]
+                    ValueFromPipelineByPropertyName=$true,
+                    ParameterSetName = 'Key')]
         [Alias('Versions')]
         [string] $Name,
 
```

Comment:
The existing remove functions require another get-* to retrieve the object again. Rather than accept the pipeline input from the object that I otherwise already have in memory. That is slow and redundant and it leveraging the object passed through the pipe. I suggest the other removes are modified to reduce the extra get-* unless needed.